engines/docker/image_cache.go

```diff
@@ -0,0 +1,108 @@
+package dockerengine
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	docker "github.com/fsouza/go-dockerclient"
+	"github.com/taskcluster/taskcluster-worker/runtime"
+	"github.com/taskcluster/taskcluster-worker/runtime/atomics"
+	"github.com/taskcluster/taskcluster-worker/runtime/caching"
+	"github.com/taskcluster/taskcluster-worker/runtime/gc"
+)
+
+const dockerPullImageInactivityTimeout = 30 * time.Second
+
+type imageResource struct {
+	Repository string
+	Tag        string
+	Size       uint64
+	client     *docker.Client
+	monitor    runtime.Monitor
+	dispose    atomics.Once
+}
+
+type cacheContext struct {
+	context.Context
+}
+
+func newCacheContext(ctx context.Context) *cacheContext {
+	if ctx != nil {
+		return &cacheContext{
+			Context: ctx,
+		}
+	}
+	return nil
+}
+
+func (ctx *cacheContext) Progress(description string, percent float64) {
+	debug("progress: %s %f", description, percent)
+}
+
+func (im *imageResource) MemorySize() (uint64, error) {
+	return im.Size, nil
+}
+
+func (im *imageResource) DiskSize() (uint64, error) {
+	return 0, gc.ErrDisposableSizeNotSupported
```

Comment:
This is where you want to return `im.Size` :)

Reply:
This was confusing, because it looked like this required the size of the disk and not the size of the resource on disk. Should we consider calling this SizeOnDisk() ?